src/Equations/elastic/Model/ElasticSetup.h

```diff
@@ -0,0 +1,219 @@
+              break;
+        
+            case 2:
+              o_M(8,0) = -i_material.lambda;
+              o_M(8,1) = -i_material.lambda;
+              o_M(8,2) = -lambda2mu;
+              o_M(7,4) = -i_material.mu;
+              o_M(6,5) = -i_material.mu;
+              o_M(2,8) = -rhoInv;
+              if (!testIfAcoustic(i_material.mu)) {
+                o_M(5,6) = -rhoInv;
+                o_M(4,7) = -rhoInv;
+              }
+              break;
+              
+            default:
+              break;
+          }
+      }
+
+    template<typename Tloc, typename Tneigh>
+      inline void getTransposedGodunovState( ElasticMaterial const& local,
+                                             ElasticMaterial const& neighbor,
+                                             FaceType               faceType,
+                                             Tloc&                  QgodLocal,
+                                             Tneigh&                QgodNeighbor )
+      {
+         QgodNeighbor.setZero();
+
+         // Eigenvectors are precomputed
+         Matrix99 R = Matrix99::Zero();
+       
+         if (testIfAcoustic(local.mu)) {
+           R(0,0) = local.lambda;
+           R(1,0) = local.lambda;
+           R(2,0) = local.lambda;
+           R(6,0) = std::sqrt((local.lambda) / local.rho);
+       
+           R(3,1) = 1.0;
+           R(5,2) = 1.0;
+         } else {
+           R(0,0) = local.lambda + 2*local.mu;
+           R(1,0) = local.lambda;
+           R(2,0) = local.lambda;
+           R(6,0) = std::sqrt((local.lambda + 2 * local.mu) / local.rho);
+       
+           R(3,1) = local.mu;
+           R(7,1) = std::sqrt(local.mu / local.rho);
+       
+           R(5,2) = local.mu;
+           R(8,2) = std::sqrt(local.mu / local.rho);
+         }
+       
+         R(4,3) = 1;
+       
+         R(1,4) = 1;
+       
+         R(2,5) = 1;
+       
+         if (testIfAcoustic(neighbor.mu)) {
+           R(7,6) = 1.0;
+           R(8, 7) = 1.0;
+       
+           R(0,8) = neighbor.lambda;
+           R(1,8) = neighbor.lambda;
+           R(2,8) = neighbor.lambda;
+           R(6,8) = -std::sqrt((neighbor.lambda + 2 * neighbor.mu) / neighbor.rho);
+         } else {
+           R(5,6) = neighbor.mu;
+           R(8,6) = -std::sqrt(neighbor.mu / neighbor.rho);
+       
+           R(3,7) = neighbor.mu;
+           R(7,7) = -std::sqrt(neighbor.mu / neighbor.rho);
+       
+           R(0,8) = neighbor.lambda + 2*neighbor.mu;
+           R(1,8) = neighbor.lambda;
+           R(2,8) = neighbor.lambda;
+           R(6,8) = -std::sqrt((neighbor.lambda + 2 * neighbor.mu) / neighbor.rho);
+         }
+       
+       
+         if (faceType == FaceType::freeSurface) {
+           getTransposedFreeSurfaceGodunovState( testIfAcoustic(local.mu), QgodLocal, QgodNeighbor, R);
+         } else {
+           Matrix99 chi = Matrix99::Zero();
+           chi(0,0) = 1.0;
+           chi(1,1) = 1.0;
+           chi(2,2) = 1.0;
```

Comment:
I don't think this is correct. You only have one P-Wave for an acoustic material. Please double check with Lukas.